### PR TITLE
spirv-fuzz: Transformations to add types, constants and variables

### PR DESCRIPTION
**Pull request by afd**
This change adds several transformations that allow types, constants,
undefined values and global variables to be added to a module.

### CODE REVIEW

File: source/fuzz/transformation_add_global_variable.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  if (message_.initializer_id()) {
+    auto initilizer_instruction =
+        context->get_def_use_mgr()->GetDef(message_.initializer_id());
+    // If an initializer id is provided it must correspond to an instruction.
+    if (!initilizer_instruction) {
+      return false;
+    }
+    // The instruction must be a constant.
+    switch (initilizer_instruction->opcode()) {
+      case SpvOpConstant:
```
Comment (paulthomson) [inline source/fuzz/transformation_add_global_variable.cpp]: Is there not also `SpvOpConstantTrue`, `SpvOpConstantNull`? In a later transformation, you use `ConstantManager::GetConstantsFromIds` to check if things are constant. Could you do that here as well?

File: source/fuzz/transformation_add_type_matrix.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  if (!fuzzerutil::IsFreshId(context, message_.fresh_id())) {
+    return false;
+  }
+  auto column_type =
+      context->get_type_mgr()->GetType(message_.column_type_id());
+  if (!column_type) {
+    return false;
+  }
+  return column_type->AsVector() &&
+         column_type->AsVector()->element_type()->AsFloat();
```
Comment (paulthomson) [inline source/fuzz/transformation_add_type_matrix.cpp]: Does it need to be a float? Maybe it obviously has to be but if not could add a comment in the proto file?

File: source/fuzz/transformation_add_global_variable.cpp (hunk 1 above)
Comment (afd) [reply source/fuzz/transformation_add_global_variable.cpp]: Good point - addressed in the manner suggested.

File: source/fuzz/transformation_add_type_matrix.cpp (hunk 2 above)
Comment (afd) [reply source/fuzz/transformation_add_type_matrix.cpp]: Yes it does - I have added comments in this file and in the .proto file to clarify that the SPIR-V spec requires this.

**Pull request merged**